File: app/src/main/java/ru/vpiska/shop/Shop.java
package ru.vpiska.shop;

import java.util.Comparator;

/**
 * Created by Кирилл on 15.12.2018.
 */


public class Shop {
    private String productId,title;
    double price;



    Shop(String productId,String title,double price){
        this.productId = productId;
        this.title = title;
        this.price = Math.round(price);
    }

    public static Comparator<Shop> ShopComparator = new Comparator<Shop>() {

        public int compare(Shop s1, Shop s2) {


            //ascending order
            return Double.compare(s1.getPrice(), s2.getPrice());

            //descending order
            //return StudentName2.compareTo(StudentName1);
        }};

    public String getProductId() {
        return productId;
    }

    public String getTitle() {
        return title;
    }

    public double getPrice() {
        return price;
    }
}

File: app/src/main/java/ru/vpiska/shop/Shop.kt
package ru.vpiska.shop

import java.util.*

/**
 * Created by Кирилл on 15.12.2018.
 */


class Shop internal constructor(val productId: String, val title: String, price: Double) {
    var price: Double = 0.toDouble()
        internal set


    init {
        this.price = Math.round(price).toDouble()
    }

    companion object {

        var ShopComparator: Comparator<Shop> = Comparator { s1, s2 ->
            //ascending order
            java.lang.Double.compare(s1.price, s2.price)

            //descending order
            //return StudentName2.compareTo(StudentName1);
        }
    }
}
